string action_str
string[] action_names
int32[] action_ids
int8 selected_action
int8 selected_step
int8 selected_arm_step